src/components/CompareResults/CompareResultsTable.tsx

```diff
@@ -6,52 +10,237 @@ import TableContainer from '@mui/material/TableContainer';
   const compareResults: CompareResultsItem[] = useAppSelector(
     (state) => state.compareResults,
   );
+  const filteredResults: CompareResultsItem[] = useAppSelector(
+    (state: RootState) => state.filterCompareResults.filteredResults,
+  );
+
+  filter.platform.options = new Set(
+    compareResults.map((result) => result.platform),
+  );
+
+  const suiteOptions = compareResults.map((result) => result.suite);
+  filter.test.options = new Set([
+    ...compareResults.map((result) => result.test),
+    ...suiteOptions,
+  ]);
+
+  const [openFilter, setOpenFilter] = React.useState<string | null>(null);
+
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
+    null,
+  );
+
+  const handleOpenOptions = (
+    event: React.MouseEvent<HTMLButtonElement>,
+    key: string,
+  ) => {
+    setOpenFilter(key);
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleCloseOptions = () => {
+    setAnchorEl(null);
+    setOpenFilter(null);
+  };
+
+  const open = Boolean(anchorEl);
+  const filterKeys = Object.keys(filter);
+
+  const results: CompareResultsItem[] = filteredResults.length
+    ? filteredResults
+    : compareResults;
+
+  const activeFilters: ActiveFilters = useAppSelector(
+    (state) => state.filterCompareResults.activeFilters,
+  );
+  const { confidence, platform, test } = activeFilters;
+  const isFiltered = !!(confidence.length || platform.length || test.length);
+
   return (
-    <TableContainer component={Paper}>
-      <Table sx={{ minWidth: 650 }} size="small" aria-label="a dense table">
-        <TableHead>
-          <TableRow>
-            {tableHeaders.map((header) => (
-              <TableCell align={header.align} key={header.id}>
-                {header.label}
-              </TableCell>
-            ))}
-          </TableRow>
-        </TableHead>
-        <TableBody>
-          {compareResults.length > 0 &&
-            compareResults.map((result, index) => (
-              <CompareResultsTableRow
-                key={index}
-                result={result}
-                index={index}
-                mode={mode}
-              />
-            ))}
-        </TableBody>
-      </Table>
-    </TableContainer>
+    <React.Fragment>
+      <Autocomplete
+        id="size-small-standard"
+        size="small"
+        sx={{ width: 200, marginBottom: 5 }}
+        options={frameworks}
+        isOptionEqualToValue={(option, value) => option.name === value.name}
+        getOptionLabel={(option) => option.name}
+        defaultValue={frameworks[0]}
+        renderInput={(params) => (
+          <TextField
+            {...params}
+            variant="standard"
+            label="Framework"
+            placeholder="Select framework"
+          />
+        )}
+      />
+      {isFiltered && <Alert severity="info">Filters have been applied</Alert>}
+      <TableContainer component={Paper}>
+        <Table sx={{ minWidth: 650 }} size="small" aria-label="a dense table">
```

Comment:
Eventually (for a future PR) it would be good to have a proper aria-label naming this table.